internal/handlers/common/msg_findandmodify.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+)
+
+// FindAndModifyParams represent all findAndModify requests' fields.
+// It's filled by calling prepareFindAndModifyParams.
+type FindAndModifyParams struct {
+	DB, Collection, Comment               string
+	Query, Sort, Update                   *types.Document
+	Remove, Upsert                        bool
+	ReturnNewDocument, HasUpdateOperators bool
+	MaxTimeMS                             int32
+}
+
+// PrepareFindAndModifyParams prepares findAndModify request fields.
+func PrepareFindAndModifyParams(document *types.Document) (*FindAndModifyParams, error) {
+	var err error
+
+	command := document.Command()
+
+	var db string
+
+	if db, err = GetRequiredParam[string](document, "$db"); err != nil {
+		return nil, err
+	}
```

Comment:
```suggestion
	db, err := GetRequiredParam[string](document, "$db")
	if err != nil {
		return nil, err
	}
```

I know `wsl` could be annoying, but we should not follow it blindly